OSDOCS#6275: Add release notes for cert-manager Operator 1.11 (#60715)
openshift/openshift-docs: security/cert_manager_operator/cert-manager-operator-release-notes.adoc

Reviewer: IIUC we should be doing the release notes in reverse chronology order?
1.11 comes first and 1.10 comes next in order, WDYT; can you also check with other operators/product if this would be suitable ordering, please.

Author: That's right. Corrected the sequence

[id="cert-manager-operator-1.11.1-bug-fixes"]
=== Bug fixes

* Previously, the `cm-acme-http-solver` pod did not use the latest published Red Hat image `registry.redhat.io/cert-manager/jetstack-cert-manager-acmesolver-rhel9`. With this release, the `cm-acme-http-solver` pod uses the latest published Red Hat image `registry.redhat.io/cert-manager/jetstack-cert-manager-acmesolver-rhel9`. (link:https://issues.redhat.com/browse/OCPBUGS-10821[*OCPBUGS-10821*])

* Previously, the {cert-manager-operator} did not support changing labels for cert-manager pods such as controller, CA injector, and Webhook pods. With this release, you can add labels to cert-manager pods. (link:https://issues.redhat.com/browse/OCPBUGS-8466[*OCPBUGS-8466*])

> * Previously, you could not update the log verbosity level in the {cert-manager-operator}. Now, you can update the log verbosity level by using an environmental variable `OPERATOR_LOG_LEVEL` in its subscription resource. (link:https://issues.redhat.com/browse/OCPBUGS-9994[*OCPBUGS-9994*])

Reviewer 2: ```suggestion
* Previously, you could not update the log verbosity level in the {cert-manager-operator}. You can now update the log verbosity level by using an environmental variable `OPERATOR_LOG_LEVEL` in its subscription resource. (link:https://issues.redhat.com/browse/OCPBUGS-9994[*OCPBUGS-9994*])
```

> * Previously, when uninstalling the {cert-manager-operator}, if you select the *Delete all operand instances for this operator* checkbox in the {product-title} web console, the Operator was not uninstalled properly. Now, the {cert-manager-operator} is now properly uninstalled. (link:https://issues.redhat.com/browse/OCPBUGS-9960[*OCPBUGS-9960*])

Reviewer 2: ```suggestion
* Previously, when uninstalling the {cert-manager-operator}, if you select the *Delete all operand instances for this operator* checkbox in the {product-title} web console, the Operator was not uninstalled properly. The {cert-manager-operator} is now properly uninstalled. (link:https://issues.redhat.com/browse/OCPBUGS-9960[*OCPBUGS-9960*])
```

> * Previously, the {cert-manager-operator} did not support using Google workload identity federation. Now, the {cert-manager-operator} supports using Google workload identity federation. (link:https://issues.redhat.com/browse/OCPBUGS-9998[*OCPBUGS-9998*])

Reviewer 2: ```suggestion
* Previously, the {cert-manager-operator} did not support using Google workload identity federation. The {cert-manager-operator} now supports using Google workload identity federation. (link:https://issues.redhat.com/browse/OCPBUGS-9998[*OCPBUGS-9998*])
```

* You can use the {cert-manager-operator} to set up an ACME issuer to solve DNS-01 challenges by using ambient and explicit credentials on GCP.

[id="cert-manager-dns01-acme-azure-1.11.1"]
==== Configuring an ACME issuer to solve DNS01 challenges on Microsoft Azure
* You can use the {cert-manager-operator} to set up an ACME issuer to solve DNS-01 challenges by using explicit credentials on Microsoft Azure.

[id="cert-manager-operator-1.11.1-bug-fixes"]
=== Bug fixes

* Previously, the `cm-acme-http-solver` pod did not use the latest published Red Hat image `registry.redhat.io/cert-manager/jetstack-cert-manager-acmesolver-rhel9`. With this release, the `cm-acme-http-solver` pod uses the latest published Red Hat image `registry.redhat.io/cert-manager/jetstack-cert-manager-acmesolver-rhel9`. (link:https://issues.redhat.com/browse/OCPBUGS-10821[*OCPBUGS-10821*])

* Previously, the {cert-manager-operator} did not support changing labels for cert-manager pods such as controller, CA injector, and Webhook pods. With this release, you can add labels to cert-manager pods. (link:https://issues.redhat.com/browse/OCPBUGS-8466[*OCPBUGS-8466*])

> * Previously, the `operatorLogLevel` field did not update the log verbosity level in the {cert-manager-operator}. Now, the `operatorLogLevel` field properly updates the log verbosity level in the {cert-manager-operator}. (link:https://issues.redhat.com/browse/OCPBUGS-9994[*OCPBUGS-9994*])

Reviewer 3: operatorLogLevel field does not work for this. OPERATOR_LOG_LEVEL does it.
```suggestion
* Previously, there is no way to update the log verbosity level in the {cert-manager-operator}. Now it is implemented using an environmental variable OPERATOR_LOG_LEVEL in its subscription resource. (link:https://issues.redhat.com/browse/OCPBUGS-9994[*OCPBUGS-9994*])
```

Author: Addressed. Changed the wording little bit

* Previously, when uninstalling the {cert-manager-operator}, if you select the *Delete all operand instances for this operator* checkbox in the {product-title} web console, the Operator was not uninstalled properly. Now, the {cert-manager-operator} is now properly uninstalled. (link:https://issues.redhat.com/browse/OCPBUGS-9960[*OCPBUGS-9960*])

* Previously, the {cert-manager-operator} did not support using Google workload identity federation. Now, the {cert-manager-operator} supports using Google workload identity federation. (link:https://issues.redhat.com/browse/OCPBUGS-9998[*OCPBUGS-9998*])

> * Previously, if you set the log level and the log level verbosity for the CA injector and Webhook components, the {cert-manager-operator} used to panic and fail to update the CA injector and Webhook pods. Now, the {cert-manager-operator} updates the CA injector and Webhook pods with the log level and log level verbosity.  (link:https://issues.redhat.com/browse/OCPBUGS-14129[*OCPBUGS-14129*])

Reviewer 3: Need not mention [OCPBUGS-14129](https://issues.redhat.com/browse/OCPBUGS-14129). Because "previously" released 1.10.2 does not have this bug. It is only introduced during 1.11.1.

> * Previously, after installing the {cert-manager-operator}, the {product-title} web console showed `Installed operator - ready for use` message before the cert-manager controller, CA injector, and Webhook pods became available. (link:https://issues.redhat.com/browse/OCPBUGS-9954[*OCPBUGS-9954*])

Reviewer 3: Need not mention OCPBUGS-9954 because it is converted to a doc bug and seemed already documented.

* You can use {cert-manager-operator} to set up an ACME issuer to solve DNS-01 challenges by using explicit credentials on Microsoft Azure.

[id="cert-manager-operator-1.11.1-bug-fixes"]
=== Bug fixes

* Previously, the `cm-acme-http-solver` pod did not use the latest published Red Hat image `registry.redhat.io/cert-manager/jetstack-cert-manager-acmesolver-rhel9`. With this release, the `cm-acme-http-solver` pod uses the latest published Red Hat image `registry.redhat.io/cert-manager/jetstack-cert-manager-acmesolver-rhel9`. (link:https://issues.redhat.com/browse/OCPBUGS-10821[*OCPBUGS-10821*])

* Previously, the {cert-manager-operator} did not support changing labels for cert-manager pods such as controller, CA injector, and Webhook pods. With this release, you can add labels to cert-manager pods. (link:https://issues.redhat.com/browse/OCPBUGS-8466[*OCPBUGS-8466*])

* Previously, the `operatorLogLevel` field did not update the log verbosity level in the {cert-manager-operator}. Now, the `operatorLogLevel` field properly updates the log verbosity level in the {cert-manager-operator}. (link:https://issues.redhat.com/browse/OCPBUGS-9994[*OCPBUGS-9994*])

> 

Reviewer: More bug fixes we needa add to this list:
1. OCPBUGS-9960
2. OCPBUGS-9998
3. OCPBUGS-14129
4. OCPBUGS-9954
